docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+
+## Table of Contents
+
+The playbook is divided into three sections:
+
+- **Supporting the New Architecture in your Library**
+  - Prerequisites for Supporting the New Architecture in JavaScript
+  - Enabling the New Architecture in your Library
+    - Android, iOS
+- **Supporting the New Architecture in your App**
+  - Prerequisites for Supporting the New Architecture in your App
+  - Enabling the New NativeModule System (TurboModule) in your App
+    - Android, iOS
+  - Enabling the New Renderer (Fabric) in your App
+    - Android, iOS
+- **Appendix**
+
+# Supporting the New Architecture in Your Library
+
+## Prerequisites for Supporting the New Architecture in JavaScript
+
+The following steps will help ensure your modules and components are ready for the new architecture.
+
+### TurboModules: Define Specs in JavaScript
+
+Under the TurboModule system, the JavaScript spec will serve as the source of truth for the methods that are provided by each native module. Without the JavaScript spec, it is up to you to ensure your public method signatures are equivalent on Android and iOS.
+
+As the first step to adopting the new architecture, you will start by creating these specs for your native modules. You can do this, right now, prior to actually migrating your native module library to the new architecture. Your JavaScript spec will be used later on to generate native interface code for all the supported platforms, as a way to enforce uniform APIs across platforms.
+
+#### Writing the JavaScript Spec
+
+The JavaScript spec defines all APIs that are provided by the native module, along with the types of those constants and functions. You will be using Flow to type the input arguments and outputs of your native module’s methods.
+
+By convention, JavaScript spec files are named `Native<MODULE_NAME>.js` and they export a `TurboModuleRegistry` `Spec` object. The following is a basic JavaScript spec template:
+
+```javascript
+'use strict';
+
+import type {TurboModule} from 'react-native/Libraries/TurboModule/RCTExport';
+import {TurboModuleRegistry} from 'react-native';
+
+export interface Spec extends TurboModule {
+  +getConstants: () => {||};
+
+  // your module methods go here, for example:
+  +getString(id: string): Promise<string>;
+}
+
+export default (TurboModuleRegistry.get<Spec>('<MODULE_NAME>'): ?Spec);
+```
+
+#### Supported Flow Types
+
+You will be using [Flow type annotations](https://flow.org/en/docs/types/) to define your JavaScript spec. Keeping in mind that the goal of defining a JavaScript spec is to ensure the generated native interface code is type safe, the set of supported Flow types will be those that can be mapped one-to-one to a corresponding type on the native platform.
```

Comment:
>You will be using [Flow type annotations](https://flow.org/en/docs/types/) to define your JavaScript spec.

Similarly, "you will be using Flow" sounds prescriptive if TS version exists. (Does it?)

Reply:
it does (?) at least if we are taking about `react-native-tscodegen` package/comand;
as far as my tests go it works (eg generates code from typescript interfaces) but I never tested actually build in real app.

Reply:
also also (and I am really sorry if it sound harsh) prioritising flow documentation over typescript is kinda... strange.

I do understand that flow is still used internally by Facebook (Meta).

But outside, in open source react/react-native ecosystem it's almost exclusively TS - I even struggle to recall any recent packages that used flow.

Reply:
I agree with you @romanonthego, and i also think the priority should for typescript since the vast majority of the community using it

Reply:
Thank you for the feedback! We definitely want to prioritize what's useful for the open source community. Our priority was to be transparent and share an early progress of this playbook. Since Meta is using flow internally, using Flow in this playbook was one of the fastest way to showcase what it could look like to adopt the new architecture so that we can gather early feedback. For example, this feedback is very helpful to gauge how important TS is for the community (though we expected this as well).

Reply:
I've reworded a bit to sound less prescriptive + the TypeScript banner just above should also help clarify what's the current status.